optimum/neuron/modeling_seq2seq.py

```diff
@@ -0,0 +1,912 @@
+            axis=1,
+        )
+
+        # copy the new cache state to the decoder
+        for state, tensor in zip(self.decoder.model.parameters(), past_key_values):
+            state.copy_(tensor)
+
+        output = super().generate(
+            **inputs,
+            generation_config=generation_config,
+            logits_processor=logits_processor,
+            stopping_criteria=stopping_criteria,
+            prefix_allowed_tokens_fn=prefix_allowed_tokens_fn,
+            assistant_model=assistant_model,
+            streamer=streamer,
+            num_return_sequences=num_return_sequences,
+            max_length=max_length,
+            num_beams=num_beams,
+            do_sample=kwargs.pop("do_sample", False),
+            use_cache=kwargs.pop("use_cache", True),
+            decoder_attention_mask=decoder_attention_mask,
+            # Pass fake encoder_outputs so the transfomers code will not invoke the encoder
+            encoder_outputs={"last_hidden_state": torch.ones((batch_size, max_length, 1))},
+        )
+        return output
+
+    def beam_search(
+        self,
+        input_ids: torch.LongTensor,
+        beam_scorer: "BeamScorer",
+        logits_processor: Optional[LogitsProcessorList] = None,
+        stopping_criteria: Optional[StoppingCriteriaList] = None,
+        pad_token_id: Optional[int] = None,
+        eos_token_id: Optional[Union[int, List[int]]] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        output_scores: Optional[bool] = None,
+        return_dict_in_generate: Optional[bool] = None,
+        synced_gpus: Optional[bool] = False,
+        seq_length: Optional[int] = None,
+        **model_kwargs,
+    ) -> Union[BeamSearchOutput, torch.LongTensor]:
+        """
+        Overriding beam search to use next_token_scores returned from neuron device instead of logits.
+        """
+        logits_processor = logits_processor if logits_processor is not None else LogitsProcessorList()
+        stopping_criteria = stopping_criteria if stopping_criteria is not None else StoppingCriteriaList()
+        pad_token_id = pad_token_id if pad_token_id is not None else self.generation_config.pad_token_id
+        eos_token_id = eos_token_id if eos_token_id is not None else self.generation_config.eos_token_id
+        if isinstance(eos_token_id, int):
+            eos_token_id = [eos_token_id]
+        output_scores = output_scores if output_scores is not None else self.generation_config.output_scores
+        output_attentions = (
+            output_attentions if output_attentions is not None else self.generation_config.output_attentions
+        )
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.generation_config.output_hidden_states
+        )
+
+        batch_size = len(beam_scorer._beam_hyps)
```

Comment:
You might want to check that `beam_scorer` has actually instantiated, because the default value is a `str`.

Reply:
How can it be a str? According to the typing, it should be an instance of `BeamScorer` right? I just saw we did the same in `NeuronGenerationMixin`.

https://github.com/huggingface/optimum-neuron/blob/50ec86f387f0220c2f890749d7b776169c34c7da/optimum/neuron/generation/utils.py#L266

Reply:
Hummm. My mistake. I mistook the type hint to a default value in the signature of your method